blog/20230219-best-8-message-queue/best-8-message-queue.md

```diff
@@ -0,0 +1,175 @@
+
+* Cloud becomes the infrastructure of society: Digitalization has become an important driving force driving the development of enterprises.More and more enterprises choose to build their digital business based on public cloud.In the 10 years from 2010 to 2020, the global cloud computing market size ranged from $41 billion to $312 billion. Even during the epidemic, the market growth rate in 2020 will still be as high as 33%.
+* Global economy enters recession: Although we don't want to see it, we have to admit that the global economy is now in a massive recession. The spread of the epidemic around the world is one of the most important reasons. The recession has made business extremely difficult. Saving costs has become an important topic for many business executives.
+* Cloud native is becoming more and more popular: Modern enterprises demand better agility and flexibility and lower costs from their digital businesses. This has given birth to the rapid development of cloud native technology. For example, C I/CD technology can provide the ability of rapid delivery, and Serverless technology can provide the ability of fast elasticity and on-demand operation.
+* kubertenes is becoming the infra of cloud native apps：Kubernetes provides a powerful ability to automatically expand and shrink applications, and dynamically adjust resources according to the load of the application, thereby achieving higher resource utilization and faster application response time.his helps enterprises save costs and improve efficiency, so more and more enterprises deploy their software on k8s.
+
+Around 2010, due to the surge of mobile Internet users, a large amount of data needs to be processed, which gave birth to the emergence of message queues such as Kafka. In 2020, due to the large number of cloud technologies adopted by enterprises and the emergence of cloud-native technologies such as kubernetes and serverless, enterprises have new needs. This time, they need a message queue with a cloud-native architecture that is truly suitable for the new infrastructure. However, message queues, which were born around 2010, are obviously powerless in the face of new infrastructure and new applications such as serverless due to different technical architectures and application scenarios. For example, kafka obviously has many problems running on kubertenes:
+
+* StatefulSet requirement: Kafka is a distributed system that requires each node to maintain state, which can make it difficult to run on Kubernetes. In particular, running Kafka on Kubernetes requires using StatefulSets, which can be more complex to manage than Deployments.
+
+* Resource consumption: Kafka requires significant resources to run, including CPU, memory, and storage. This can make it challenging to run Kafka in a scalable way on Kubernetes, where resources are typically shared among many different applications.
```

Comment:
```suggestion
* Resource consumption: Kafka requires significant resources to run, including CPU, memory, and storage. This can make it challenging to run Kafka in a scalable way on Kubernetes, where resources are typically shared among many different applications.
```